src/Microsoft.AspNetCore.WebUtilities/FileBufferingReadStream.cs

```diff
@@ -135,22 +180,27 @@ private Stream CreateTempFile()
         public override int Read(byte[] buffer, int offset, int count)
         {
             ThrowIfDisposed();
-            if (_buffer.Position < _buffer.Length || _completelyBuffered)
+            if (_buffer.Position < _readIntoBuffer || _completelyBuffered)
```

Comment:
Ah, it assumes the `byte[]` started full of data. You could call `SetLength(0)` just after creating the stream to reset it.